src/main/java/org/semux/config/AbstractConfig.java

```diff
@@ -52,10 +52,10 @@
     // =========================
     // Chain spec
     // =========================
-    protected long maxBlockGasLimit = 10_000_000L; // 10m gas
+    protected long maxBlockGasLimit = 15_000_000L; // 10m gas
     protected Amount minTransactionFee = MILLI_SEM.of(5);
     protected Amount minDelegateBurnAmount = SEM.of(1000);
-    protected long nonVMTransactionGasCost = 21_000L;
+    protected long nonVMTransactionGasCost = 5_000L;
```

Comment:
This should prob still be 21,000.  Keep gas scaled proportionate.

Reply:
I'd like to have `21000` as well but block 64 has more than 2000 transactions https://semux.info/explorer/block/65

Reply:
Yes. We still need to figure out a reasonable gas limit.  Just one of my sample erc20 creates uses 2 mill gas...  Had to bump up cost from old VM testnet for some reason. Not sure if bug or bugfix.  So 15mil may be quite low

Reply:
@semuxgo I think even 5_000L is not enough. 1 month ago someone submitted 3237 TXs in one block. This one seems to be the biggest block of mainnet so far: https://semux.info/explorer/block/1405616

```
 height  | transactionsCount
---------+-------------------
 1405616 |              3237
 1399832 |              2941
      66 |              2483
 1405617 |              2338
      65 |              2276
 1405593 |              1734
 1405494 |              1694
  353180 |              1604
  353176 |              1593
   99479 |              1287
```

Reply:
I'm okay with bumping the block gas limit to resolve this issue. Currently, Ethereum has a gas limit of ~8mil. Don't want to go too high to avoid any unforeseen troubles.

Reply:
Also, I'm going to keep the 5000 gas for non-vm-transaction and the rational is that they can be executed more efficiently without launching a virtual machine. This would allow higher throughput without worrying about vm denial-of-service.